spec/features/adventure_spec.rb

```diff
@@ -0,0 +1,186 @@
+        fill_in "user_password", with: "abc123"
+        fill_in "user_password_confirmation", with: "abc123"
+        click_on "Sign up"
+
+        expect(page).to have_content("You have signed up successfully")
+      end
+    end
+    context "when updating an account's details" do
+      before do
+        login_as(user, :scope => :user)
+      end
+
+      it "does not let you change your password if you don't know your current password" do
+        visit edit_user_registration_path
+        fill_in "user_password", with: "newpass"
+        fill_in "user_password_confirmation", with: "newpass"
+        fill_in "user_current_password", with: "notthepassword"
+        click_on "Update"
+
+        expect(page).to have_content("Current password is invalid")
+      end
+
+      it "does not let you change your password if password confirmation doesn't match" do
+        visit edit_user_registration_path
+        fill_in "user_password", with: "newpass"
+        fill_in "user_password_confirmation", with: "newpasswrong"
+        fill_in "user_current_password", with: "password"
+        click_on "Update"
+
+        expect(page).to have_content("Password confirmation doesn't match Password")
+      end
+
+      it "allows you to change your password if everything is correct" do
+        visit edit_user_registration_path
+        fill_in "user_password", with: "newpass"
+        fill_in "user_password_confirmation", with: "newpass"
+        fill_in "user_current_password", with: "password"
+        click_on "Update"
+
+        expect(page).to have_content("Your account has been updated successfully.")
+        expect(page).to have_current_path("/")
+      end
+    end
+    context "when logged in and accessing stories" do
+      let!(:adventure)          { create(:adventure, user: user) }
+
+      before do
+        login_as(user, :scope => :user)
+      end
+
+      context "when dealing with password-protected stories" do
+        let!(:adventure_password) { create(:adventure, user: user, slug: "test-story-password", has_password: true, password: "password") }
+
+        it "allows a creator to view a password protected story without entering password", js: true do
+          visit adventure_path(adventure_password)
+
+          expect(page).to have_content("Test Story")
+          expect(page).to have_content("begin")
+          expect(page).to have_content("EDIT")
+        end
+
+      end
+
+      it "lets you view your stories" do
+        visit "/mine"
+
+        expect(page).to have_content(adventure.description)
+      end
+
+      it "lets you edit your story" do
+        visit details_adventure_path(adventure)
+
+        expect(page).to have_content("What's your title?")
+        expect(page).to have_content("Story URL")
+      end
+
+      it "does not allow you to update a story with a blank title" do
+        adventure.update_attributes(title: nil)
+
+        expect(adventure.save).to eq(false)
+      end
```

Comment:
The code here could be rewritten as `it { should validate_presence_of :title }` in a model spec (which ... doesn't exist yet). Looks like the spec below this is the proper looking feature spec for ensuring that invalid form submissions are gracefully handled.